src/engine/SCons/Node/Python.py

```diff
@@ -96,6 +96,10 @@ def __init__(self, value, built_value=None):
         if built_value is not None:
             self.built_value = built_value
 
+        # Set a name so it can be a child of a node and not break
+        # its parent's implementation of Node.get_contents.
+        self.name = value
```

Comment:
which get_contents in Node/__init__.py references a python value nodes name?

Reply:
@bdbaddog get_contents_dir uses it as a sorting key. See https://github.com/SCons/scons/blob/master/src/engine/SCons/Node/__init__.py#L216